beacon-chain/db/kv/blocks_test.go

```diff
@@ -50,6 +50,50 @@ func TestStore_BlocksCRUD(t *testing.T) {
 	}
 }
 
+func TestStore_BlocksBatchDelete(t *testing.T) {
+	db := setupDB(t)
+	defer teardownDB(t, db)
+	ctx := context.Background()
+	blocks := []*ethpb.BeaconBlock{
+		{
+			Slot:       1,
+			ParentRoot: []byte("parent"),
+		},
+		{
+			Slot:       2,
+			ParentRoot: []byte("parent"),
+		},
+	}
+	if err := db.SaveBlocks(ctx, blocks); err != nil {
+		t.Fatal(err)
+	}
+	retrieved, err := db.Blocks(ctx, filters.NewFilter().SetParentRoot([]byte("parent")))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(retrieved) != 2 {
+		t.Errorf("Received %d blocks, wanted 2", len(retrieved))
+	}
+	roots := make([][32]byte, len(blocks))
+	for i := 0; i < len(roots); i++ {
+		r, err := ssz.SigningRoot(blocks[i])
+		if err != nil {
+			t.Fatal(err)
+		}
+		roots[i] = r
+	}
+	if err := db.DeleteBlocks(ctx, roots); err != nil {
+		t.Fatal(err)
+	}
+	retrieved, err = db.Blocks(ctx, filters.NewFilter().SetParentRoot([]byte("parent")))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(retrieved) > 0 {
```

Comment:
This tests all or nothing, perhaps we should add a test case for being in the middle. [b0,b1,b2,b3,b4] and delete all the even blocks... etc

Reply:
Same with attestation as well

Reply:
Good idea, will add those for better cov

Reply:
Made it so we delete the even items and verify only the odd items remain